spec/lib/whitehall_importer/embed_body_references_spec.rb

```diff
@@ -12,31 +12,135 @@
       expect(govspeak_body).to eq("[Contact:#{content_id}]")
     end
 
-    it "converts Whitehall image syntax to Content Publisher syntax" do
-      govspeak_body = described_class.call(
-        body: "!!1 test !!2",
-        images: ["foo.png", "bar.jpg"],
-      )
+    context "embedded images" do
+      it "does not convert two Windows carridge returns" do
+        govspeak_body = described_class.call(
+          body: "\r\n\r\n!!1",
+          images: ["file.jpg"],
+        )
 
-      expect(govspeak_body).to eq("[Image:foo.png] test [Image:bar.jpg]")
-    end
+        expect(govspeak_body).to eq("\r\n\r\n[Image:file.jpg]")
+      end
 
-    it "removes any image markdown that doesn't resolve to an image" do
-      govspeak_body = described_class.call(
-        body: "Bar !!2 Baz",
-        images: ["foo.png"],
-      )
+      it "does not convert two Unix carridge returns" do
+        govspeak_body = described_class.call(
+          body: "\n\n!!1",
+          images: ["file.jpg"],
+        )
+
+        expect(govspeak_body).to eq("\n\n[Image:file.jpg]")
+      end
+
+      it "converts one Windows carridge return" do
+        govspeak_body = described_class.call(
+          body: "\r\n!!1",
+          images: ["file.jpg"],
+        )
+
+        expect(govspeak_body).to eq("\r\n\r\n[Image:file.jpg]")
+      end
+
+      it "converts one Unix carridge return" do
+        govspeak_body = described_class.call(
+          body: "\n!!1",
+          images: ["file.jpg"],
+        )
 
-      expect(govspeak_body).to eq("Bar  Baz")
+        expect(govspeak_body).to eq("\n\n[Image:file.jpg]")
+      end
+
+      it "converts no carridge returns" do
+        govspeak_body = described_class.call(
+          body: "!!1",
+          images: ["file.jpg"],
+        )
+
+        expect(govspeak_body).to eq("[Image:file.jpg]")
+      end
+
+      it "converts Whitehall image syntax to Content Publisher syntax" do
+        govspeak_body = described_class.call(
+          body: "!!1 test !!2",
+          images: ["file.png", "file.jpg"],
+        )
+
+        expect(govspeak_body).to eq("[Image:file.png] test !!2")
+      end
+
+      it "removes any image markdown that doesn't resolve to an image" do
+        govspeak_body = described_class.call(
+          body: "Bar !!2 Baz",
+          images: ["file.png"],
+        )
+
+        expect(govspeak_body).to eq("Bar !!2 Baz")
+      end
     end
 
-    it "converts Whitehall attachment syntax to Content Publisher syntax" do
-      govspeak_body = described_class.call(
-        body: "!@1 test !@2",
-        attachments: ["file.pdf", "download.csv"],
-      )
+    context "embedded attachments" do
```

Comment:
Seems a bit of a bummer that we repeat all the same tests again - hmm we could try a shared example but that will have quite a lot of arguments so probably may cause more harm than good.

Maybe we can combine things to try make them more concise.

For example:

```
it "converts Whitehall image bang embeds to govspeak Image embeds" do
  body = described_class.call(body: "!!1", images: ["foo.png"])
  expect(body).to eq("[Image:foo.png]")
end

it "converts Whitehall attachment bang embeds to govspeak Image embeds" do
  body = described_class.call(body: "!@1", attachments: ["foo.pdf"])
  expect(body).to eq("[Attachment:foo.pdf]")
end

it "converts image and attachment embeds prefixed by single new lines to be prefixed by two" do
  body = described_class.call(body: "\n!!1\n!@1", images: ["foo.png"], attachments: ["foo.pdf"])
  expect(body).to eq("\n\n[Image:foo.png]\n\n[Attachment:foo.pdf]")
  body = described_class.call(body: "\r\n!!1\r\n!@1", images: ["foo.png"], attachments: ["foo.pdf"])
  expect(body).to eq("\r\n\r\n[Image:foo.png]\r\n\r\n[Attachment:foo.pdf]")
end

it "doesn't change the prefixes of images and attachments preceeded by two or more new lines" do
  body = described_class.call(body: "\n\n!!1\n\n\n!@1", images: ["foo.png"], attachments: ["foo.pdf"])
  expect(body).to eq("\n\n[Image:foo.png]\n\n\n[Attachment:foo.pdf]")

  body = described_class.call(body: "\r\n\r\n\r\n!!1\r\n\r\n!@1", images: ["foo.png"], attachments: ["foo.pdf"])
  expect(body).to eq("\r\n\r\n\r\n[Image:foo.png]\r\n\r\n[Attachment:foo.pdf]")
end
```

Reply:
Umm, I'm not a big fan of munging concepts like this, but I see your point!

I've had a go - let me know if I've DRYed it too far.